### PR TITLE
[5.2] [PHP8.4] Update for nullable type declaration

### PR DESCRIPTION
**Pull request by heelc29**
### Summary of Changes

updates for https://wiki.php.net/rfc/deprecate-implicitly-nullable-types

### Testing Instructions

~~at the moment no QA release of PHP8.4 ... so only code review (and phpcs by drone)~~ [PHP 8.4.0 Alpha 1 available for testing](https://www.php.net/archive/2024.php#2024-07-05-1)

### Actual result BEFORE applying this Pull Request

Warning:

```
Deprecated: Joomla\CMS\Application\AdministratorApplication::__construct(): Implicitly marking parameter $input as nullable is deprecated, the explicit nullable type must be used instead in libraries\src\Application\AdministratorApplication.php on line 74

Deprecated: Joomla\CMS\Application\AdministratorApplication::__construct(): Implicitly marking parameter $config as nullable is deprecated, the explicit nullable type must be used instead in libraries\src\Application\AdministratorApplication.php on line 74

Deprecated: Joomla\CMS\Application\AdministratorApplication::__construct(): Implicitly marking parameter $client as nullable is deprecated, the explicit nullable type must be used instead in libraries\src\Application\AdministratorApplication.php on line 74

Deprecated: Joomla\CMS\Application\AdministratorApplication::__construct(): Implicitly marking parameter $container as nullable is deprecated, the explicit nullable type must be used instead in libraries\src\Application\AdministratorApplication.php on line 74

[...]
```

### Expected result AFTER applying this Pull Request

No new deprecated warnings (excepted composer dependencies) if update to PHP8.4

### Link to documentations
Please select:
- [x] No documentation changes for docs.joomla.org needed
- [x] No documentation changes for manual.joomla.org needed

### CODE REVIEW

**Comment by HLeithner** [comment]
This pull request has been automatically rebased to 5.2-dev.

File: libraries/src/Document/Document.php [hunk 1]
```diff
@@ -325,7 +325,7 @@ public function __construct($options = [])
         if (\array_key_exists('webAssetManager', $options)) {
             $this->setWebAssetManager($options['webAssetManager']);
         } else {
-            $webAssetManager = new WebAssetManager(\Joomla\CMS\Factory::getContainer()->get('webassetregistry'));
+            $webAssetManager = new WebAssetManager(CmsFactory::getContainer()->get('webassetregistry'));
```
Comment (HLeithner) [inline libraries/src/Document/Document.php:328]: That's not a nullable declaration change but ok for me, I checked the `use` statement.

**Review by HLeithner — APPROVED** [review]
I did a code review on all 286 changed files, all of them looking ok except one which is also ok but not changing implizit `null`
image

File: libraries/src/Document/Document.php (hunk 1 above)
Comment (laoneo) [reply libraries/src/Document/Document.php:328]: Thanks

**Comment by laoneo** [comment]
Thanks!

**Pull request merged**